src/Compilers/VisualBasic/Portable/Semantics/Conversions.vb

```diff
@@ -916,13 +916,16 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             userDefinedConversionsMightStillBeApplicable = False
             Dim conv As ConversionKind
 
-            ' Using <symbol>.IsConstant() for field accesses can result in an infinite loop.
+            ' Using source.IsConstant for field accesses can result in an infinite loop.
             ' To detect such a loop pass the already visited constants from the binder.
+            ' The check for source.IsConstant is still necessary because the node might still 
+            ' be considered as a non-constant in some error conditions (a reference before declaration,
+            ' for example).
```

Comment:
All source.IsConstant does is checking if source.ConstantValueOpt is not Nothoing. If sourceIsConstant is true, the code below (ClassifyNothingLiteralConversion, ClassifyNumericConstantConversion) is going to request ConstantValueOpt anyway, so adding source.IsConstant condition at the end with AndAlso doesn't add any new chances for an infinite loop.
Regarding the comment. I think the comment is quite clear. However, if you provide a complete wording that you believe is better, I will happily replace the comment with it.